types/react-native-auth0/index.d.ts

```diff
@@ -182,39 +180,56 @@ export class Auth {
     multifactorChallenge(params: MultiFactorChallengeParams): Promise<MultiFactorChallengeResponse>;
 }
 
+export interface Auth0Connection {
+    user_id: string;
+    provider: string;
+    connection: string;
+    isSocial: boolean;
+}
+
 /**
  * Users
  */
-export interface Auth0User<T> {
+export interface Auth0User<T = void> {
     created_at: string;
-    email: string;
-    emailVerified: boolean;
-    identities: any[];
+    email?: string;
+    email_verified?: boolean;
+    identities: Auth0Connection[];
```

Comment:
As mentioned above, the identities structure wouldn't have particular definition.